Matt's branch merge into master

mattmanb opened the pull request:
I began adding **methods** to a lot of the different classes. _**ESPECIALLY board.js!!!!**_ Make sure to look over my code before pulling this to master (I could've made a _mistake_). ALSO - refer to "Notes/Notes on the project.txt" for some other methods we need to implement (if I do them I will update the notes)

mattmanb committed:
started some work on board.js

mattmanb committed:
Added several methods to several classes.
Lots of work on board.js class.
Thinking about how we're gonna query the user
if they want to buy the property/utility/railroad.
LOOK AT "Notes/Notes on the project.txt" for other
methods we should implement

mattmanb committed:
fixed small things, redoing pull request

mattmanb committed:
chance_card constructor was inside chancedeck on
accident

mattmanb committed:
Removed jail.js class (we can handle jail in
the player class), also began integrating code
from monopoly.js into backend.js and class
methods (still have a little bit left to
transfer)

JavierCrespoLaw reviewed (commented):
Sorry that I didn't check yesterday, was going home for Thanksgiving break. The code mostly looks good, however Railroad and Utility don't have payRent or queryPurchase functions. The functions can be empty for now, just want to make sure they exist.

mattmanb committed:
Merge branch 'master' into Matt's-Branch

mattmanb committed:
Added an avenue class to store properties.
It helps in determining if a player has a
monopoly. Also, I finished moving over code
from monopoly.js into class methods.

mattmanb committed:
fixed some bugs with "Special conditions" with
chance cards and community chest cards

mattmanb commented:
> Sorry that I didn't check yesterday, was going home for Thanksgiving break. The code mostly looks good, however Railroad and Utility don't have payRent or queryPurchase functions. The functions can be empty for now, just want to make sure they exist.

I added payRent to utility and railroad class. I'm not quite sure how im going to do 'queryPurchase' and 'startAuction' yet but I'll probably get to it over break.

JavierCrespoLaw reviewed (approved):
Nice, looks good!
